### PR TITLE
Vl53 long range

### PR DESCRIPTION
**Pull request by rspaargaren**
## Description:

**Related issue (if applicable):** fixes 
https://github.com/esphome/feature-requests/issues/724#

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#592 Will be submitted

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Review by rradar — APPROVED** [review]

**Comment by glmnet** [comment]
can you check the test yaml changes? they don't seem related to this sensor

File: tests/test2.yaml [hunk 1]
```diff
@@ -233,6 +229,25 @@ binary_sensor:
     name: "Storm Alert"
 
 esp32_ble_tracker:
+  on_ble_advertise:
```
Comment (glmnet) [inline tests/test2.yaml:232]: unrelated

File: tests/test1.yaml [hunk 2]
```diff
@@ -1044,6 +1044,14 @@ light:
               duration: 250ms
             - state: False
               duration: 250ms
+    on_turn_on:
```
Comment (glmnet) [inline tests/test1.yaml:1047]: unrelated

**Review by glmnet — CHANGES_REQUESTED** [review]

File: tests/test2.yaml (hunk 1 above)
Comment (rspaargaren) [reply tests/test2.yaml]: changes are indeed unrelated and due to github push and pulls. Removed

File: tests/test1.yaml (hunk 2 above)
Comment (rspaargaren) [reply tests/test1.yaml]: changes are unrelated and due to github push and pulls removed

**Comment by glmnet** [comment]
cc @rradar 

 I was about to ask @rspaargaren to test my change but I see here https://github.com/esphome/feature-requests/issues/724 he doesn't have the device either.

I've just pushed a commit which takes into account the signal_rate_limit, seems like it was not initialized which could cause unexpected behavior.

So if @rradar can test everything works ok (both with long_range true and false) then I'll merge this which also fixes that bug

**Comment by rspaargaren** [comment]
@glmnet I have ordered one at Ali. So couple of weeks and I can test for myself.

**Comment by rspaargaren** [comment]
> @glmnet I have ordered one at Ali. So couple of weeks and I can test for myself.

@randybb / @Svrbinek / @onlineguz

Is one of you able to test this part of the code?

**Comment by rradar** [comment]
Just tested with custom_component and the [latest code](https://github.com/rspaargaren/esphome/tree/VL53-long-range) on 1.14.5 but now I get this:

```yaml
Failed config

sensor.vl53l0x: [source vl53l0x.yaml:28]
  platform: vl53l0x
  name: ToF Distance
  update_interval: 10s
  
  [long_range] is an invalid option for [sensor.vl53l0x]. Did you mean [on_value_range]?
  long_range: True
```

compiling without long_range works like expected

**Review by rradar — COMMENTED** [review]
works like expected when set to `true` or `false`

**Review by rradar — APPROVED** [review]

**Comment by rspaargaren** [comment]
Tnx! Good to hear now I saw in other request to set the i2c adres

**Pull request merged**